Unlocked scanning was RFed in git annex, test fails

```
======================================================================
D: /a
FAIL: datalad.support.tests.test_annexrepo.test_init_scanning_message
D: /a/b/c
----------------------------------------------------------------------
D: /a~
Traceback (most recent call last):
D: /a b/name
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
testing 0
    self.test(*self.arg)
testing 1
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 739, in _wrap_with_tempfile
/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/datalad_temp_tree_test_paths_with_forward_slashes2jf0l2fq
    return t(*(arg + (filename,)), **kw)
' |;&%b5{}\'"Δקم๗あ .datc '
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_annexrepo.py", line 1311, in test_init_scanning_message
.: here(+) [git]
    assert_in("for unlocked", cml.out)
.: dl-test-remote(+) [/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/datalad_temp_test_cached_dataset07olz_ny/https___github.com_datalad_testrepo--minimalds (git)]
AssertionError: 'for unlocked' not found in ''

```

Happens with daily builds of git annex

--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -3609,9 +3609,9 @@ class AnnexInitOutput(WitlessProtocol):
     def pipe_data_received(self, fd, byts):
         line = byts.decode(self.encoding)
         if fd == 1:
-            if "scanning for unlocked files" in line:
-                lgr.info("Scanning for unlocked files "
-                         "(this may take some time)")
+            res = re.search("(scanning for .* files)", line, flags=re.IGNORECASE)
+            if res:
+                lgr.info("%s (this may take some time)", res.groups()[0])
         elif fd == 2:
             lgr.info(line.strip())
 

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1309,7 +1309,8 @@ def test_init_scanning_message(path):
         # somewhere around 8.20210428-186-g428c91606 git annex changed
         # handling of scanning for unlocked files upon init and started to report
         # "scanning for annexed" instead of "scanning for unlocked".
-        assert_re_in(".*scanning for .* files", cml.out, flags=re.IGNORECASE)
+        # Could be a line among many (as on Windows) so match=False so we search
+        assert_re_in(".*scanning for .* files", cml.out, flags=re.IGNORECASE, match=False)
 
 
 @with_tempfile

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1306,7 +1306,11 @@ def test_init_scanning_message(path):
     # | end kludge
     with swallow_logs(new_level=logging.INFO) as cml:
         AnnexRepo(path, create=True, version=7)
-        assert_in("for unlocked", cml.out)
+        # somewhere around 8.20210428-186-g428c91606 git annex changed
+        # handling of scanning for unlocked files upon init . It was optimized etc
+        # and that message is no longer appears. yoh is not yet sure when it would
+        if external_versions['cmd:annex'] <= '8.20210428':
+            assert_in("for unlocked", cml.out)
 
 
 @with_tempfile

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1307,10 +1307,9 @@ def test_init_scanning_message(path):
     with swallow_logs(new_level=logging.INFO) as cml:
         AnnexRepo(path, create=True, version=7)
         # somewhere around 8.20210428-186-g428c91606 git annex changed
-        # handling of scanning for unlocked files upon init . It was optimized etc
-        # and that message is no longer appears. yoh is not yet sure when it would
-        if external_versions['cmd:annex'] <= '8.20210428':
-            assert_in("for unlocked", cml.out)
+        # handling of scanning for unlocked files upon init and started to report
+        # "scanning for annexed" instead of "scanning for unlocked".
+        assert_re_in(".*scanning for .* files", cml.out, flags=re.IGNORECASE)
 
 
 @with_tempfile

